manifests/plugin/gutterball/config.pp

```diff
@@ -0,0 +1,17 @@
+# gutterball config
+class katello::plugin::gutterball::config(
+    $foreman_user  = 'foreman',
+    $foreman_group = 'foreman',
+    $foreman_plugins_dir = '/etc/foreman/plugins',
+    $protocol = 'https',
```

Comment:
I still think a single URL parameter is easier than all the separate components. Now you're still forced to use the fqdn which a user might not want.